### PR TITLE
Bump botocore to 1.27.0

### PR DESCRIPTION
**Pull request by gmsantos**
### Description of Change

Continuation of #952, bump botocore to 1.27.0.

For a more straightforward review, here are [the commits related to the bump only](https://github.com/aio-libs/aiobotocore/pull/953/files/43bd6299beae274e685eed2777c9618c9362e585..ee34afb40a191f5cf6e412299f5082a363a864d0).

### Checklist for All Submissions
* [x] I have added change info to [CHANGES.rst](https://github.com/aio-libs/aiobotocore/blob/master/CHANGES.rst)
* [ ] If this is resolving an issue (needed so future developers can determine if change is still necessary and under what conditions) (can be provided via link to issue with these details):
  * [ ] Detailed description of issue
  * [ ] Alternative methods considered (if any)
  * [ ] How issue is being resolved
  * [ ] How issue can be reproduced 
* [ ] If this is providing a new feature  (can be provided via link to issue with these details):
  * [ ] Detailed description of new feature
  * [ ] Why needed
  * [ ] Alternatives methods considered (if any)

### Checklist when updating botocore and/or aiohttp versions

* [x] I have read and followed [CONTRIBUTING.rst](https://github.com/aio-libs/aiobotocore/blob/master/CONTRIBUTING.rst#how-to-upgrade-botocore)
* [x] I have updated test_patches.py where/if appropriate (also check if no changes necessary)
* [x] I have ensured that the awscli/boto3 versions match the updated botocore version

-----
[View rendered .github/pull_request_template.md](https://github.com/gmsantos/aiobotocore/blob/bump-botocore-1.27.0/.github/pull_request_template.md)

### CODE REVIEW

**Comment by thehesiod** [comment]
@gmsantos needs merge/rebase from master and could you document the steps you performed to ensure we pulled all the relevant changes?  Here are the steps I usually perform: https://github.com/aio-libs/aiobotocore/blob/master/CONTRIBUTING.rst#how-to-upgrade-botocore

**Comment by thehesiod** [comment]
btw may want to bump again

**Comment by gmsantos** [comment]
> @gmsantos needs merge/rebase from master and could you document the steps you performed to ensure we pulled all the relevant changes? Here are the steps I usually perform: https://github.com/aio-libs/aiobotocore/blob/master/CONTRIBUTING.rst#how-to-upgrade-botocore

Sure, how I did was:

- first, I applied the same codestyle of botocore in aio botocore.
- from botocore I checkout both the previous release (1.24.21) and the target release (1.27.0)
- In the 1.24.21, I reapplied the same codestyle in botocore to reduce the noise from code diff
- After that, I used [meld](https://meldmerge.org/) to do a 3-way comparison between:
  - botocore 1.24.21 (patched with codestyle)
  - aiobotocore (after #952)
  - botocore 1.27.0
- I checked the changes in the source file and create a single commit for each one of the new changes from botocore 1.27.0
  - [Add _register_eventbridge_events to client](https://github.com/aio-libs/aiobotocore/pull/953/commits/cb0bda10272af309e69c9837418051dd57ba636f)
  - [Add new ec2_credential_refresh_window to create_credential_resolver](https://github.com/aio-libs/aiobotocore/pull/953/commits/e643537aa0e2570665b9bc254c783e30eb8ec33f)
  - [Changes in utils file](https://github.com/aio-libs/aiobotocore/pull/953/commits/ee34afb40a191f5cf6e412299f5082a363a864d0)

Apart from that, I bumped the requirements and updated the patches files with the new expected hashes (I had to update practically all files impacted by code style changes). 

To test in, I was able to install it on an Airflow deployment with [astronomer-providers](https://github.com/astronomer/astronomer-providers), and I didn't find any bug. Assuming that the automated tests are passing as well, it LGTM.

> btw may want to bump again

That's my plan for the following contributions. Since my primary use for aiobotocore is as a dependency on astronomer-providers (see https://github.com/aio-libs/aiobotocore/issues/924), I was planning to bump to compatible botocore releases locked in Airflow constraints.

- [Airflow 2.3.2 constraints](https://raw.githubusercontent.com/apache/airflow/constraints-2.3.2/constraints-3.10.txt) locks botocore in 1.27.0 (this PR)
- [Airflow 2.3.3 constraints](https://raw.githubusercontent.com/apache/airflow/constraints-2.3.3/constraints-3.10.txt) locks botocore in 1.27.23
- Latest botocore version is 1.27.34

If that's okay, I can contribute with some PRs to the following two bumps, to 1.27.23 in aiobotocore v2.4.1 and 1.27.34 in aiobotocore v2.4.2

**Comment by thehesiod** [comment]
thanks!  I think I can get to this today

File: aiobotocore/client.py [hunk 1]
```diff
@@ -206,7 +209,14 @@ def _register_legacy_retries(self, client):
...
-            'needs-retry.%s' % service_event_name, handler, unique_id=unique_id
+            f"needs-retry.{service_event_name}", handler, unique_id=unique_id
+        )
+
+    def _register_eventbridge_events(
+        self, client, endpoint_bridge, endpoint_url
+    ):
+        super()._register_eventbridge_events(
+            client, endpoint_bridge, endpoint_url
         )
```
Comment (thehesiod) [inline aiobotocore/client.py:220]: I don't think this needs to be added, the impl from the super should be fine given nothing is being overridden

**Comment by thehesiod** [comment]
missing `yield from` change in `eventstream.py::EventStream::_create_raw_event_generator`

**Comment by thehesiod** [comment]
missing httpsession.py changes

**Comment by thehesiod** [comment]
missing `yield from` changes from `paginate.py`

**Comment by thehesiod** [comment]
looks like str got erroneously turned into a tuple here: https://github.com/aio-libs/aiobotocore/blob/88a43098b1194bf759581acc710ad5bdbdd99a96/aiobotocore/paginate.py#L137

**Comment by thehesiod** [comment]
the new `StreamingBody.__enter__` + `StreamingBody.__exit__` in response.py need to be overridden to do the right thing.

**Comment by thehesiod** [comment]
missing change to `utils.py::InstanceMetadataFetcher::retrieve_iam_role_credentials`

**Comment by thehesiod** [comment]
btw I updated the recommended steps on that rst.  Basically I usually diff old/new versions of botocore and apply changes, then diff aiobotocore to target version to ensure they match.  Using Diffmerge it's then easy to apply changes from LHS to RHS at the end

File: aiobotocore/client.py (hunk 1 above)
Comment (gmsantos) [reply aiobotocore/client.py:220]: Got it, so calling `super()` from create_client is fine.

**Comment by LOC333** [comment]
Hey guys, wondering if you have a sense of when this will be merged? Running into dependency conflicts that this bump would resolve 😄

**Comment by gmsantos** [comment]
@LOC333 maybe soon I can take a look at the PR comments. Busy weeks... can't promise though. 

Take in mind that I'm not the maintainer, so if you are in a rush you can install from [this unreleased branch](https://stackoverflow.com/questions/20101834/pip-install-from-git-repo-branch). Again, no promises that will work as is for you.

**Comment by thehesiod** [comment]
sorry for delay guys, things have just been nuts here

**Comment by thehesiod** [comment]
done in https://github.com/aio-libs/aiobotocore/pull/963

**Pull request closed without merging**